core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -513,6 +523,16 @@ public SliceBuilder slice() {
         return sliceBuilder;
     }
 
+
+    public CollapseBuilder collapse() {
```

Comment:
I noticed that the `collapse` field wasn't taken into account in hascode, equals and swallowCopy methods.
Can you maybe alter `SearchSourceBuilderTests.java` to also take into account an `CollapseBuilder` instance?